app/models/manager.rb

```diff
@@ -11,15 +11,18 @@
 #  updated_at :datetime         not null
 #  program_id :bigint
 #
-class ProgramManager < ApplicationRecord
-  has_and_belongs_to_many :programs
+class Manager < ApplicationRecord
+  has_many :managers_programs
+  has_many :programs, through: :managers_programs
+
+  validates :uniqname, uniqueness: true
 
   def instructor
```

Comment:
move this to the Program model and call it would decouple this code

Reply:
Sorry, I don't understand what exactly to move to the Program model. Lines 15-18? Line 20?

Reply:
I was referring to (lines 20 - 30):
`def instructor
    Program.where(instructor: self)
  end

  def manager
    Program.includes(:managers).where(managers_programs: [self])
  end
`

Reply:
@rsmoke  These are methods to get programs where a uniqname is an instructor or a manager.
These methods belong to the Manager model. They can be called like this:

`Manager.last.instructor `- returns a list of programs where the Manager.last is an instructor
`Manager.last.manager `- returns a list of programs where the Manager.last is a manager

Maybe the names of these methods are not clear - any suggestions on how to name them?

Reply:
@rsmoke I keep thinking about these methods. They will be essential in the faculty authorization process. And if they must be moved to a different place that should be done in a different ticket.